### PR TITLE
Tickets/dm 5909

### PR DESCRIPTION
**Pull request by jgates108**

### CODE REVIEW

File: admin/tools/docker/3_build-git-image.sh [hunk 1]
```diff
@@ -60,10 +60,10 @@ if [ -z "$GIT_REF" -a -z "$SRC_DIR" ]; then
...
     usage
     exit 2
-elif [ -n "$GIT_REF" -a -n "$SRC_DIR" ]; then
-    echo "ERROR: must not provide both branch/tag and local source directory"
-    usage
-    exit 2
+#elif [ -n "$GIT_REF" -a -n "$SRC_DIR" ]; then
+#    echo "ERROR: must not provide both branch/tag and local source directory"
+#    usage
+#    exit 2
```
Comment (jgates108) [inline admin/tools/docker/3_build-git-image.sh]: I'll uncomment this before merging.

File: core/modules/util/EventThread.h [hunk 2]
```diff
@@ -207,6 +207,8 @@ class CommandThreadPool : public CommandTracked {
     CommandThreadPool() {}
     CommandThreadPool(std::function<void(CmdData*)> func) : Command{func} {}
 
+    PoolEventThread* getPoolEventThread() { return _poolEventThread; } //< Not thread safe.
+
     friend class PoolEventThread; // Only class that should use _poolEventThread.
```
Comment (andy-slac) [inline core/modules/util/EventThread.h]: This `friend` isn't needed any more if you add method above?

File: core/modules/rproc/InfileMerger.cc [hunk 3]
```diff
@@ -280,11 +169,17 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
             return false;
         }
     }
-    return _importResponse(response);
+
+    // Nothing to do if size is zero.
+    if (response->result.row_size() != 0) {
```
Comment (andy-slac) [inline core/modules/rproc/InfileMerger.cc]: :nitpick: (damn, no emoticon for nitpick) I'd probably move this check into `_merge()` to make it more generic.

File: core/modules/util/EventThread.h (hunk 2 above)
Comment (jgates108) [inline core/modules/util/EventThread.h]: The comment is incorrect. I'll fix it.

File: core/modules/rproc/InfileMerger.cc (hunk 3 above)
Comment (jgates108) [inline core/modules/rproc/InfileMerger.cc]: I'm removing _merge in DM-5910 and putting the code here as there's no longer any need to keep it separate with the end of the Mgr class.

**Pull request merged**